### PR TITLE
iOS: Misc tooling updates

### PR DESCRIPTION
**Pull request by chinmaygarde**
Use the project name to initialize the bundle name and identifier

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
Does this affect `flutter trace` as well?  `flutter stop`?  Seems a bit scary to land right before the hackathon, but depends on @abarth's plans for the branch.

**Comment by chinmaygarde** [comment]
> Does this affect `flutter trace` as well? `flutter stop`?

If they use the ID specified in the IOSApp instance, then yes.

File: packages/flutter_tools/lib/src/application_package.dart [hunk 1]
```diff
@@ -81,13 +82,25 @@ class AndroidApk extends ApplicationPackage {
...
-    String localPath: _defaultPath,
-    String id: _defaultId
-  }) : super(localPath: localPath, id: id);
+    String iosProjectDir,
+    String iosProjectBundleId
+  }) : super(localPath: iosProjectDir, id: iosProjectBundleId);
+
+  factory IOSApp.fromBuildConfiguration(BuildConfiguration config) {
+    if (getCurrentHostPlatform() != HostPlatform.mac) {
+      return null;
```
Comment (abarth) [inline packages/flutter_tools/lib/src/application_package.dart]: Is it legal to return null from a factory?

Comment (chinmaygarde) [inline packages/flutter_tools/lib/src/application_package.dart]: From a factory, it should be legal. It's a tiny bit odd that factories look like constructors in Dart. So I am not sure if this is idiomatic.

**Comment by chinmaygarde** [comment]
I have added another patch to this commit:
In case the user changes the path of the Flutter project on disk, the Dart SDK or the Flutter repository itself, the old Xcode paths would be invalid. Now, we rewrite them when we build the project.

File: ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
+_Detailed issue description here. More information at [https://flutter.io/bug_reports/](https://flutter.io/bug_reports/)_
```
Comment (abarth) [inline ISSUE_TEMPLATE.md]: Did you mean to include this file in this patch?

**Comment by abarth** [comment]
LGTM

File: ISSUE_TEMPLATE.md (hunk 2 above)
Comment (chinmaygarde) [inline ISSUE_TEMPLATE.md]: Eh, no. Things got backed up over the past couple of days. Do you want me to separate this into another PR?

Comment (abarth) [inline ISSUE_TEMPLATE.md]: Let's send it separately.
